app/add_some_abilities/add_potion_abilities.py

```diff
@@ -0,0 +1,9 @@
+def add_ability_from_potion(knight: dict, ability: str) -> None:
+    if check_presence_ability(knight, ability):
```

Comment:
The function `check_presence_ability` checks if the ability is in `knight['potion']['effect']`, but it should check if the ability is in `knight` to determine if the knight already has that ability.